### ISSUE TITLE
source/app.d(59,3): Error: undefined identifier updateMethods

### ISSUE THREAD

**Issue by WolfieWerewolf** [skoppe/spasm#6]
Hi... ok so I wanted to use the webidl generator to produce stubs for something simple, 'getElementById'.  I tried to run it as you instructed and got the error above... even if I just run dub in the webidl folder I get the same.. where is updateMethods() coming from?
Thanks.

**Comment by skoppe** [comment]
updateMethods comes from the open-methods library I am using.

I think it worked on my machine because it was included transitively via `import webidl.bindings.generator`, but I believe that has been deprecated in the latest D compilers (I am still on ldc 1.13).

I added it directly in `app.d` and pushed to master.

**Comment by WolfieWerewolf** [comment]
dank u wel.. that works fine now (I spent a year in Amsterdam.. well Schiphol-Rijk but close enough)

**Comment by WolfieWerewolf** [comment]
And to be fair to the work you have done.. it works incredibly well.. this webidl binding generator is fantastic...

**Comment by skoppe** [comment]
Thanks. It took some time before I figured out how it ought to work. At first I just dumped all generated js code, even for functions that aren't called. Then I released I can leverage the wasm linker to 'guide' binding generation. You see, the linker only emits imports for functions it doesn't have, and only those which are used in your code. So now I just parse the wasm binary and look for imports coming from spasm.bindings, and then generate the required js glue code (+encoders/decoders).
